### PR TITLE
Add `State.get_run`

### PR DESCRIPTION
**Pull request by panh99**
- Added `State.get_run`.
- Extended `State.creat_run`.
- Implemented `SqliteState.get_run` and `InMemoryState.get_run`.
- Passed `fab_id="None/None", fab_version="None"` to `State.create_run` in `DriverServicer`.
- Updated unit tests accordingly.

### CODE REVIEW

File: src/py/flwr/server/superlink/state/state.py [hunk 1]
```diff
@@ -150,9 +150,26 @@ def get_nodes(self, run_id: int) -> Set[int]:
         """
 
     @abc.abstractmethod
-    def create_run(self) -> int:
+    def create_run(self, fab_id: str, fab_version: str) -> int:
         """Create one run."""
```
Comment (danieljanes) [inline src/py/flwr/server/superlink/state/state.py:154]: Should we update the docstring here?

File: src/py/flwr/server/superlink/state/state.py [hunk 2]
```diff
@@ -150,9 +150,26 @@ def get_nodes(self, run_id: int) -> Set[int]:
...
+        ----------
+        run_id : int
+            The identifier of the run.
+
+        Returns
+        -------
+        Tuple[str, str]
+            A tuple containing two strings:
+            - `fab_id`: The identifier of the Flower Fabric used in the specified run.
+            - `fab_version`: The version of the Flower Fabric used in the specified run.
```
Comment (danieljanes) [inline src/py/flwr/server/superlink/state/state.py:170]: Let's add the `run_id` here as well. We can use a tuple for now, but we should probably switch to a dataclass `Run` in the future.

File: src/py/flwr/server/superlink/state/state.py (hunk 1 above)
Comment (panh99) [reply src/py/flwr/server/superlink/state/state.py]: Sure, I've updated.

File: src/py/flwr/server/superlink/state/state.py (hunk 2 above)
Comment (panh99) [reply src/py/flwr/server/superlink/state/state.py]: I am okay with adding `run_id` here, but just a little bit confused. To me, it seems like `get_run` is the method that retrieving the details of a run given its ID, so naturally the definition of the method would be ID -> details. May I know why we also return run_id?

File: src/py/flwr/server/superlink/state/state.py [hunk 3]
```diff
@@ -150,8 +150,26 @@ def get_nodes(self, run_id: int) -> Set[int]:
...
+        run_id : int
+            The identifier of the run.
+
+        Returns
+        -------
+        Tuple[int, str, str]
+            A tuple containing three elements:
+            - `run_id`: The identifier of the run, same as the specified `run_id`.
+            - `fab_id`: The identifier of the Flower Fabric used in the specified run.
+            - `fab_version`: The version of the Flower Fabric used in the specified run.
```
Comment (danieljanes) [inline src/py/flwr/server/superlink/state/state.py:171]:
```suggestion
            - `fab_id`: The identifier of the Flower App Bundle used in the specified run.
            - `fab_version`: The version of the Flower App Bundle used in the specified run.
```

File: src/py/flwr/server/superlink/state/state_test.py [hunk 4]
```diff
@@ -43,6 +43,19 @@ def state_factory(self) -> State:
...
+        """Test if create_run and get_run work correctly."""
+        # Prepare
+        state: State = self.state_factory()
+        run_id = state.create_run("Mock/mock", "v1.0.0")
+
+        # Execute
+        _, fab_id, fab_version = state.get_run(run_id)
+
+        # Assert
+        assert fab_id == "Mock/mock"
```
Comment (danieljanes) [inline src/py/flwr/server/superlink/state/state_test.py:56]:
```suggestion
        # Execute
        actual_run_id, fab_id, fab_version = state.get_run(run_id)

        # Assert
        assert actual_run_id == run_id
        assert fab_id == "Mock/mock"
```

**Review by danieljanes — APPROVED** [review]

File: src/py/flwr/server/superlink/state/state.py (hunk 2 above)
Comment (danieljanes) [reply src/py/flwr/server/superlink/state/state.py]: You're right, we could equally well not do this. There's no strong reason for doing this. It's more of a convenience thing. When the `Run` object gets more complex in the future, we can just return the full object and then serialize it and send it back to the SuperNode.

File: src/py/flwr/server/superlink/state/state.py [hunk 5]
```diff
@@ -150,8 +150,26 @@ def get_nodes(self, run_id: int) -> Set[int]:
...
+            The identifier of the run.
+
+        Returns
+        -------
+        Tuple[int, str, str]
+            A tuple containing three elements:
+            - `run_id`: The identifier of the run, same as the specified `run_id`.
+            - `fab_id`: The identifier of the Flower App Bundle used in the specified run.
+            - `fab_version`: The version of the Flower App Bundle used in the specified run.
+        """
```
Comment (danieljanes) [inline src/py/flwr/server/superlink/state/state.py:172]:
```suggestion
            - `fab_id`: The identifier of the FAB used in the specified run.
            - `fab_version`: The version of the FAB used in the specified run.
        """
```

**Review by danieljanes — APPROVED** [review]

**Pull request merged**